io-sim/src/Control/Monad/IOSim.hs

```diff
@@ -375,6 +375,8 @@ instance MonadAsync (SimM s) where
   cancel a@(Async tid _) = throwTo tid AsyncCancelled <* waitCatch a
   cancelWith a@(Async tid _) e = throwTo tid e <* waitCatch a
 
+  asyncWithUnmask k = async (k unblock)
```

Comment:
Doesn't it need to use forkWithUnmask, since it needs to start the thread masked right?

Reply:
Hmm, you're right. this is wrong.  Let me fix this.

Reply:
So turns out you're right :-)

Reply:
So this is correct because
> The new thread inherits the masked state of the parent (see mask).
http://hackage.haskell.org/package/base-4.14.0.0/docs/Control-Concurrent.html#v:forkIO
So it's ok to use a normal async and pass unblock.